portal-impl/src/com/liferay/portal/service/impl/RepositoryLocalServiceImpl.java

```diff
@@ -57,7 +57,11 @@ public Repository addRepository(
 
 		Repository repository = repositoryPersistence.create(repositoryId);
 
-		repository.setUuid(serviceContext.getUuid());
+		String uuid = serviceContext.getUuid();
+
+		repository.setUuid(uuid);
+		repository.setExternalReferenceCode(uuid);
```

Comment:
I think it is not necessary to set the external reference code in this case. See `RepositoryPersistenceImpl` around line 2859 in this same PR (in the autogen commit).